rust/src/lib/ifaces/inter_ifaces_controller.rs

```diff
@@ -234,3 +270,25 @@ pub(crate) fn set_ifaces_up_priority(ifaces: &mut Interfaces) -> bool {
     }
     ret
 }
+
+pub(crate) fn find_unknown_type_port<'a>(
+    iface: &'a Interface,
+    cur_ifaces: &Interfaces,
+) -> Vec<&'a str> {
+    let mut ret: Vec<&str> = Vec::new();
+    if let Some(port_names) = iface.ports() {
+        for port_name in port_names {
+            if let Some(port_iface) =
+                cur_ifaces.get_iface(port_name, InterfaceType::Unknown)
+            {
+                if port_iface.iface_type() == InterfaceType::Unknown {
+                    ret.push(port_name);
+                }
+            } else {
+                // Remove not found interface also
+                ret.push(port_name);
+            }
+        }
+    }
+    ret
```

Comment:
This whole function could be simplified using `flat_map`, but leaving it up to you.

Reply:
Unless big benefit, I intend to avoid these magics.
The cargo clippy message https://rust-lang.github.io/rust-clippy/master/index.html#option_map_unit_fn remind me why Perl been dropped by community(too many magic tricks)